gui/include/autoconnecthelper.h

```diff
@@ -0,0 +1,155 @@
+#ifndef SSIDHELPER_H
```

Comment:
Why do this by shelling out? The Chiaki subcommands themselves (or rather the code that implements them) can just be called via C++.

I'm sure there are OS-specific APIs that can be used to get the SSID of the currently connected WiFi AP as well (NetworkManager's dbus API has the info e.g.), but it's probably a lot more code to make it truly cross-platform, so it's probably ok to shell out for that.

Reply:
I looked into calling the shell commands but they just return an int status code of the command result rather than any objects I can infer the console's from. I think, from my rudimentary understanding, they all handle the discovery asynchronously via callbacks. Though now I've written that down I'm not sure it's right. I think it uses callbacks but it's all synchronous as the terminal waits for a response. Let me take a look when I get a minute and see if I can sort it.